modin/core/execution/dispatching/factories/factories.py

```diff
@@ -427,58 +427,42 @@ def _to_parquet(cls, *args, **kwargs):
 
-
-@doc(_doc_factory_class, execution_name="cuDFOnRay")
-class CudfOnRayFactory(BaseFactory):
-    @classmethod
-    @doc(_doc_factory_prepare_method, io_module_name="``cuDFOnRayIO``")
-    def prepare(cls):
-        from modin.core.execution.ray.implementations.cudf_on_ray.io import cuDFOnRayIO
-
-        cls.io_cls = cuDFOnRayIO
-
-
-@doc(_doc_factory_class, execution_name="PandasOnRay")
-class PandasOnRayFactory(BaseFactory):
-    @classmethod
-    @doc(_doc_factory_prepare_method, io_module_name="``PandasOnRayIO``")
-    def prepare(cls):
-        from modin.core.execution.ray.implementations.pandas_on_ray.io import (
-            PandasOnRayIO,
-        )
-
-        cls.io_cls = PandasOnRayIO
-
-
-@doc(_doc_factory_class, execution_name="PandasOnPython")
-class PandasOnPythonFactory(BaseFactory):
+    # experimental methods that don't exist in pandas
     @classmethod
-    @doc(_doc_factory_prepare_method, io_module_name="``PandasOnPythonIO``")
-    def prepare(cls):
-        from modin.core.execution.python.implementations.pandas_on_python.io import (
-            PandasOnPythonIO,
-        )
-
-        cls.io_cls = PandasOnPythonIO
-
+    @doc(
+        _doc_io_method_raw_template,
+        source="CSV files",
+        params=_doc_io_method_kwargs_params,
+    )
+    def _read_csv_glob(cls, **kwargs):
+        current_execution = get_current_execution()
+        if current_execution not in supported_execution:
+            raise NotImplementedError(
+                f"`_read_csv_glob()` is not implemented for {current_execution} execution."
+            )
+        return cls.io_cls.read_csv_glob(**kwargs)
 
-@doc(_doc_factory_class, execution_name="PandasOnDask")
-class PandasOnDaskFactory(BaseFactory):
     @classmethod
-    @doc(_doc_factory_prepare_method, io_module_name="``PandasOnDaskIO``")
-    def prepare(cls):
-        from modin.core.execution.dask.implementations.pandas_on_dask.io import (
-            PandasOnDaskIO,
-        )
-
-        cls.io_cls = PandasOnDaskIO
-
+    @doc(
+        _doc_io_method_raw_template,
+        source="Pickle files",
+        params=_doc_io_method_kwargs_params,
+    )
+    def _read_pickle_distributed(cls, **kwargs):
+        current_execution = get_current_execution()
+        if current_execution not in supported_execution:
+            raise NotImplementedError(
+                f"`_read_pickle_distributed()` is not implemented for {current_execution} execution."
+            )
+        return cls.io_cls.read_pickle_distributed(**kwargs)
 
-@doc(_doc_abstract_factory_class, role="experimental")
-class ExperimentalBaseFactory(BaseFactory):
     @classmethod
-    @_inherit_docstrings(BaseFactory._read_sql)
-    def _read_sql(cls, **kwargs):
+    @doc(
+        _doc_io_method_raw_template,
+        source="SQL files",
+        params=_doc_io_method_kwargs_params,
+    )
+    def _read_sql_distributed(cls, **kwargs):
         supported_engines = ("Ray", "Unidist", "Dask")
```

Comment:
Should we check for executions here, but not engines? Remember PyarrowOnRay.

Reply:
Fixed